test/lib/PlacesTestUtils.js

```diff
@@ -69,7 +118,28 @@ const PlacesTestUtils = Object.freeze({
         }
       );
     });
-    return (yield promise);
+
+    yield notifPromise;
+    yield insertPromise;
+
+    // poll every 10ms until history items appear on disk
+    yield waitUntil(function*() {
```

Comment:
This is a little convoluted, IMHO. You have a function that seems to return to different types. Could maybe be simplified to:

```JS
  yield waitUntil(function*() {
    let rows = yield getPlacesRows();
    if ((rows.length - initialNumRows) !== urlList.length) {
      return false;
    }
    let rowChecked = 0;
    rows.filter(
      row => urlSet.has(row.url) && row.frecency > -1 && row.title
    ).forEach(
      () => rowChecked++
    )
    return rowChecked === urlList.length;
  }, 10);
```